internal/controller/volsync/vshandler.go

```diff
@@ -207,6 +211,77 @@ func RDStatusReady(rd *volsyncv1alpha1.ReplicationDestination, log logr.Logger)
 	return true
 }
 
+func (v *VSHandler) setRDAsOwnerOfPVC(
+	rd *volsyncv1alpha1.ReplicationDestination,
+	pvc *corev1.PersistentVolumeClaim,
+) error {
+	// Only set OwnerReference if RD and PVC are in the same namespace
+	if rd.Namespace != pvc.Namespace {
+		return nil
+	}
+
+	// Work on a deep copy to avoid mutating caller's object
+	updated := pvc.DeepCopy()
+
+	// Create a new controller reference
+	ref := metav1.NewControllerRef(
+		rd,
+		volsyncv1alpha1.GroupVersion.WithKind("ReplicationDestination"),
+	)
+
+	// Overwrite OwnerReferences with the new one
+	updated.SetOwnerReferences([]metav1.OwnerReference{*ref})
+
+	// Update the PVC
+	if err := v.client.Update(v.ctx, updated); err != nil {
+		v.log.Error(err, "Failed to update ProtectedPVC", "PVC", pvc.Name)
```

Comment:
Why log this error here? we return the error so the caller is responsible for logging. This patterns mans we log the same error at least twice, and in the worst case many times. This makes it hard to debug and creates unhelpful noise in the log.

Instead of logging return the error, maybe with additional context:

    return fmt.Errorf("failed to set owner reference: %w", err)